### ISSUE TITLE
Failed to set parking heater mode to off[BUG]

### ISSUE THREAD

**Issue by lissmeister** [lendy007/homeassistant-skodaconnect#10]
## Environment

- skodaconnect release with the issue: v1.0.30-rc7
- Last working homeassistant-skodaconnect release (if known): v1.0.30-rc5
- Home Assistant Core release with the issue: core-2021.1.5
- Operating environment (Home Assistant/Supervised/Docker/venv): Home Assistant OS 5.10
- Car model and year: Super 2018
- Valid We Connect subscription: Yes
- Debug logs enabled: Yes

## Describe the bug

When attempting to deactivate the preheater an error occurs stating that there is an invalid or no response and then failure to execute.

## Steps to Reproduce

Toggle the switch for disabling the pre heater.

## Expected behavior

It should be deactivated

## Screenshots

![Screenshot 2021-02-01 at 10 46 37](https://user-images.githubusercontent.com/2478763/106441834-d529b480-647a-11eb-8856-80b1df612e8b.png)
Some additional log ocurrances too

## Traceback/Error logs

```txt

DEBUG (MainThread) [skodaconnect.connection] HTTP POST "https://msg.volkswagen.de/fs-car/bs/rs/v1/skoda/CZ/vehicles/vin-goes-here/action"
ERROR (MainThread) [skodaconnect.connection] Failure to execute:
WARNING (MainThread) [skodaconnect.vehicle] Failed to set parking heater mode to off - Invalid or no response
  File "/config/custom_components/skodaconnect/switch.py", line 38, in async_turn_off
  File "/usr/local/lib/python3.8/site-packages/skodaconnect/dashboard.py", line 530, in turn_off
  File "/usr/local/lib/python3.8/site-packages/skodaconnect/vehicle.py", line 407, in set_pheater

```

**Comment by Farfar** [comment]
This was definitely working in RC5?
And start preheater works in RC7?
Then it's a simple diff to see what I changed that might have botched it. Always tricky with preheater since it's "legacy" API and I can't test it myself.

**Comment by lissmeister** [comment]
@Farfar Just reinstalled RC5 and it doesn't work there either, which makes me wonder if it really did work before.

I´m pretty sure that it did work then since i wrote that in a comment before the weekend. https://github.com/lendy007/homeassistant-skodaconnect/issues/8#issuecomment-769163737

**Comment by Farfar** [comment]
The error handling changed the last couple of RC releases since it was a hassle trying to get exceptions carried correctly and informative. Could be som typo somewhere.
It's interesting though this part:

> ERROR (MainThread) [skodaconnect.connection] Failure to execute:

Since the error is blank there's an exception but no message which is weird. It might be that a HTTP error occured since there's not, or isn't a need for, as extensive error handling for set functions as with get. It's enough to know if set succeeded or failed.

What might be an issue is that the SPIN token is generated and sent for both start and stop pre heater, it was previously only for start but preventing it from being generated for 'stop' would need more checks and I didn't think it was necessary.

Do you have access to the skodaconnect/connection.py file? If so you can change line 829 from:
```
            self._session_headers['x-mbbSecToken'] = await self.get_sec_token(vin = vin, spin = spin, action = 'heating')
```
to:
```
            if not 'quickstop' in data:
                self._session_headers['x-mbbSecToken'] = await self.get_sec_token(vin = vin, spin = spin, action = 'heating')
```

and report back if it works?

**Comment by lissmeister** [comment]
That seems to work! While trying i somehow made too many requests and ended up getting 429 - too many requests.

The app basically told me to take the car for a ride before any more requests would be allowed.
![image](https://user-images.githubusercontent.com/2478763/106515143-84937500-64d5-11eb-97af-1c05abe3c70c.png)

```
ERROR (MainThread) [skodaconnect.connection] Unhandled HTTP exception: 429, message='Too Many Requests', url=URL('https://msg.volkswagen.de/fs-car/bs/vsr/v1/skoda/CZ/vehicles/vin-goes-here/requests')
WARNING (MainThread) [skodaconnect.vehicle] Failed to execute data refresh - Invalid or no response
  File "/config/custom_components/skodaconnect/switch.py", line 32, in async_turn_on
  File "/usr/local/lib/python3.8/site-packages/skodaconnect/dashboard.py", line 370, in turn_on
  File "/usr/local/lib/python3.8/site-packages/skodaconnect/vehicle.py", line 486, in set_refresh
```

Maybe should also declare `request_state = ''` on line 843 of `connection.py` or remove the variable altogether from the dict in line 845. HA was complaining when trying to access the non-existing variable.

**Comment by Farfar** [comment]
> That seems to work! While trying i somehow made too many requests and ended up getting 429 - too many requests.
> 
> The app basically told me to take the car for a ride before any more requests would be allowed.
> ![image](https://user-images.githubusercontent.com/2478763/106515143-84937500-64d5-11eb-97af-1c05abe3c70c.png)
> 
> ```
> ERROR (MainThread) [skodaconnect.connection] Unhandled HTTP exception: 429, message='Too Many Requests', url=URL('https://msg.volkswagen.de/fs-car/bs/vsr/v1/skoda/CZ/vehicles/vin-goes-here/requests')
> WARNING (MainThread) [skodaconnect.vehicle] Failed to execute data refresh - Invalid or no response
>   File "/config/custom_components/skodaconnect/switch.py", line 32, in async_turn_on
>   File "/usr/local/lib/python3.8/site-packages/skodaconnect/dashboard.py", line 370, in turn_on
>   File "/usr/local/lib/python3.8/site-packages/skodaconnect/vehicle.py", line 486, in set_refresh
> ```
> 
> Maybe should also declare `request_state = ''` on line 843 of `connection.py` or remove the variable altogether from the dict in line 845. HA was complaining when trying to access the non-existing variable.

Excellent. Yes, that variable is a remnant I've forgotten to remove. I will incorportate this in next release which will probably be 1.0.30 final.

**Comment by Farfar** [comment]
Closing this issue as it is solved with 1.0.30
